packages/react-dom/src/__tests__/ReactBrowserEventEmitter-test.js

```diff
@@ -0,0 +1,556 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let React;
+let ReactDOM;
+let ReactTestUtils;
+let ReactBrowserEventEmitter;
+let LISTENER = jest.fn();
+
+let createInitialInstance;
+
+let updateInstance;
+
+let CHILD, PARENT;
+
+describe('ReactBrowserEventEmitter', () => {
+  let container;
+
+  beforeEach(() => {
+    jest.resetModules();
+    LISTENER.mockClear();
+
+    React = require('react');
+    ReactDOM = require('react-dom');
+    ReactBrowserEventEmitter = require('react-dom/src/events/ReactBrowserEventEmitter');
```

Comment:
This is still a private API. The point of this rewrite is to remove reliance on private APIs.